jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/content/InputStreamContentSource.java

```diff
@@ -59,17 +78,17 @@ public Content.Chunk read()
 
         try
         {
-            byte[] buffer = new byte[getBufferSize()];
-            int read = inputStream.read(buffer);
+            ByteBuffer buffer = bufferPool.acquire(getBufferSize(), false);
+            int read = inputStream.read(buffer.array());
```

Comment:
Not sure you can do this... Shouldn't `arrayOffset`, etc. be used?

Reply:
technically yes, but currently no buffer allocated by our pools will have a non zero arrayOffset.  But I will add.